plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -496,6 +501,172 @@ private void predicatePushdownTest(String oracleType, String oracleLiteral, Stri
         }
     }
 
+    @Test
+    public void testJoinPushdownWithImplicitCast()
+    {
+        try (TestTable leftTable = new TestTable(getQueryRunner()::execute, "left_table", "(id int, varchar_50 varchar(50))", ImmutableList.of("(1, 'India')", "(2, 'Poland')"));
+                TestTable rightTable = new TestTable(getQueryRunner()::execute, "right_table_", "(varchar_100 varchar(100), varchar_unbounded varchar)", ImmutableList.of("('India', 'Japan')", "('France', 'Poland')"))) {
+            String leftTableName = leftTable.getName();
+            String rightTableName = rightTable.getName();
+            Session session = joinPushdownEnabled(getSession());
+
+            // Implict cast between bounded varchar
+            assertJoinFullyPushedDown(session, leftTableName, rightTableName, "LEFT JOIN", "l.varchar_50 = r.varchar_100", asList(1, 2));
```

Comment:
could you please asList -> List.of everywhere. So it looks the same and does not catch the attention.